modules/scripts/startup-script/files/install_monitoring_agent.sh

```diff
@@ -79,7 +94,13 @@ handle_redhat() {
 	}
 
 	install_opsagent() {
-		curl -s https://dl.google.com/cloudagents/add-google-cloud-ops-agent-repo.sh | bash -s -- --also-install
+		curl -s "${OPSAGENT_SCRIPT_URL}" | bash -s -- --also-install
```

Comment:
A couple broad comments on this script:

1. I think it would benefit from `set -e -o pipefail`
2. I'm dubious generally of piping from `curl` to `bash` without intermediate error checking. I realize that is a lot of changes in this script but a failed curl (e.g. no internet access) will pipe something to bash.
3. is_legacy_installed() seems like it has an `||` when it probably ought to have an `&&` relationship. I understand the power comes from having both legacy agents installed. That said, I see how it would complicate the script to enforce that level of detail. Thoughts? Maybe add a comment?

Reply:
1: done
2: discussed offline. Good to leave as is since error does not generate stdout and pipefail will catch error
3: I have updated logic to cover this case.